numpy/lib/index_tricks.py

```diff
@@ -769,8 +769,8 @@ def fill_diagonal(a, val, wrap=False):
     a : array, at least 2-D.
       Array whose diagonal is to be filled, it gets modified in-place.
 
-    val : scalar
-      Value to be written on the diagonal, its type must be compatible with
+    val : scalar or array_like optional
```

Comment:
```suggestion
    val : scalar or array_like
```
`optional` is intended to signify when a keyword argument is optional, so it doesn't belong here.